### DISCUSSION TITLE
Quartz scheduler time interval should increase if Job takes more time in execution. (in Asp.net Core))

### DISCUSSION THREAD

**Discussion by tomarvineet** [quartznet/quartznet#1654] [Q&A]
Hi, Suppose we have created one job that completes execution within 20 minutes and runs on a time interval of 10 minutes, but if job takes more time (10 minutes more) in execution, then time interval should adjust automatically.

Ex - Suppose a job starts at 10:00AM and completes execution at 10:20AM. Again job will start @10:30AM with the interval of 10 minutes. If Job takes 10 more minutes in execution and completes execution @10:30AM, then again job should start @10:40AM rather than @10:30AM.
Schedule time should be increased according to time taken in execution (It should be changed dynamically.)

**Comment by lahma** [comment]
This is what misfire instructions are for. You can decide how overlapping jobs behave. Also check the DisallowConcurrentExecutionAttribute.

**Reply by tomarvineet** [reply]
Thanks for your reply. I'm new to Quartz. Do you have any example for this?

**Reply by lahma** [reply]
Well there's some documentation at least:

* https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/more-about-triggers.html#misfire-instructions
* https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/more-about-jobs.html#job-state-and-concurrency

**Reply by tomarvineet** [reply]
Thanks, but still didn't find the solution to update scheduled interval time automatically if there is delay in job execution.
